twisted/internet/inotify.py

```diff
@@ -137,6 +137,7 @@ def _notify(self, filepath, events):
         """
         Callback function used by L{INotify} to dispatch an event.
         """
+        filepath = filepath.asBytesMode()
```

Comment:
Ok, cool, let's just only return bytes mode paths and document both on the relevant functions and the `INotify` class that while text mode paths are accepted, they are converted to bytes mode paths internally and that's what will be returned to users.